Emerald Insight.js

```diff
@@ -0,0 +1,240 @@
+			return "bookSection";
+		}
+		else return "journalArticle";
+	}
+	else if (getSearchResults(doc, url, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+
+function getSearchResults(doc, url, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows;
+	if (url.includes("insight/search?")) {
+		// search results
+		rows = doc.querySelectorAll('h2>a.intent_link');
+	}
+	else {
+		// journal issue
+		rows = doc.querySelectorAll('.intent_issue_item h4>a');
+		if (!rows.length) {
+			// book
+			rows = doc.querySelectorAll('li.intent_book_chapter>a');
+		}
+	}
+	for (let i = 0; i < rows.length; i++) {
+		let href = rows[i].href;
+		let title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, url, false), function (items) {
+			if (!items) {
+				return;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var DOI = url.match(/\/(10\.[^#?/]+\/[^#?/]+)\//)[1];
+	var risURL = "/insight/content/doi/" + DOI + "/full/ris";
+	// Z.debug(risURL);
+	var pdfURL = attr(doc, 'a.intent_pdf_link', 'href');
+	// Z.debug("pdfURL: " + pdfURL);
+	ZU.doGet(risURL, function (text) {
+		// they number authors in their RIS...
+		text = text.replace(/A\d+\s+-/g, "AU  -");
+		// add a comma after the last name
+		text = text.replace(/AU\s\s-\s(\w+)/g, "AU  - $1,");
+		// Z.debug(text);
+		
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		translator.setString(text);
+		translator.setHandler("itemDone", function (obj, item) {
+			if (pdfURL) {
+				item.attachments.push({
+					url: pdfURL,
+					title: "Full Text PDF",
+					mimeType: "application/pdf"
+				});
+			}
+			else {
+				item.attachments.push({
+					title: "Snapshot",
+					document: doc
+				});
+			}
+			item.complete();
+		});
+		translator.translate();
+	});
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://www.emerald.com/insight/content/doi/10.1108/IJPH-07-2016-0028/full/html",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "Universal opt-out screening for hepatitis C virus (HCV) within correctional facilities is an effective intervention to improve public health",
+				"creators": [
+					{
+						"lastName": "Morris",
+						"firstName": "Meghan D.",
+						"creatorType": "author"
+					},
+					{
+						"lastName": "Brown",
+						"firstName": "Brandon",
+						"creatorType": "author"
+					},
+					{
+						"lastName": "Allen",
+						"firstName": "Scott A.",
+						"creatorType": "author"
+					}
+				],
+				"date": "January 1, 2017",
```

Comment:
Should we make this language independent with `ZU.strToISO`? (Maybe even in the RIS translator itself?)

Reply:
I'll impement this here -- let's think through some RIS scenarios together. Generally sounds like we should.